### ISSUE TITLE
the apache-2.0.txt header is not used (I think) in the eclipse template/checkstyle

### ISSUE THREAD

**Issue by bavardage** [palantir/gradle-baseline#29]
How about we change from regexp checkstyle to something like

```
<module name="Header">
    <property name="headerFile" value="${samedir}/../copyright/apache-2.0.txt"/>
    <property name="fileExtensions" value=".java,.ts"/>
</module>
```

and then I guess also make sure that the eclipse template uses it?

**Comment by uschi2000** [comment]
I think I had tried this originally; this is not compatible with how IntelliJ interprets this file, I believe: checkstyle expects comments (// or /\* */), IntelliJ does not. Maybe the right solution is to remove the IntelliJ-specific stuff and move to the gradle license plugin (https://github.com/hierynomus/license-gradle-plugin) ?

**Comment by markelliot** [comment]
I think the bug here is that you're using an Intellij feature to check the license in Intellij but Checkstyle to check the license in Gradle and Eclipse. If you swap Intellij to use the Checkstyle feature, things should be okay?

**Comment by markelliot** [comment]
Or maybe not? I can't find any references to apache-2.0.txt, so it appears to be completely unused. How does intellij decide to use this file?

**Comment by uschi2000** [comment]
The baseline-idea plugin adds an XML config entry for every file in the .baseline/copyright/ directory.

**Comment by markelliot** [comment]
Cool, so could we quit doing that and have IntelliJ use Checkstyle to enforce instead of its header enforcement? (this makes for overall better consistency, IMO)

**Comment by uschi2000** [comment]
IntelliJ already uses checkstyle to enforce the existence. The copyright file here is used to generate the header when creating a new class. The equivalent eclipse configuration is in gradle-baseline-java-config/resources/eclipse/static/dotfile.settings/org.eclipse.jdt.ui.prefs :

```
14 org.eclipse.jdt.ui.text.custom_code_templates=<?xml version\="1.0" encoding\="UTF-8" standalone\="no"?><templates><template autoinsert\="true" context\="gettercomment_context" deleted\="false" description\="Comment for getter method" enabled\="true" id\="org.eclipse.jdt.ui.text.codetemplates.gettercomment" name\="gettercomment">/**\n * @return the ${bare_fi     eld_name}\n */</template><template autoinsert\="true" context\="settercomment_context" deleted\="false" description\="Comment for setter method" enabled\="true" id\="org.eclipse.jdt.ui.text.codetemplates.settercomment" name\="settercomment">/**\n * @param ${param} the ${bare_field_name} to set\n */</template><template autoinsert\="true" context\="constructo     rcomment_context" deleted\="false" description\="Comment for created constructors" enabled\="true" id\="org.eclipse.jdt.ui.text.codetemplates.constructorcomment" name\="constructorcomment">/**\n * ${tags}\n */</template><template autoinsert\="false" context\="filecomment_context" deleted\="false" description\="Comment for created Java files" enabled\="true"      id\="org.eclipse.jdt.ui.text.codetemplates.filecomment" name\="filecomment"/><template autoinsert\="true" context\="typecomment_context" deleted\="false" description\="Comment for created types" enabled\="true" id\="org.eclipse.jdt.ui.text.codetemplates.typecomment" name\="typecomment">/**\n * ${tags}\n */</template><template autoinsert\="false" context\="     fieldcomment_context" deleted\="false" description\="Comment for fields" enabled\="true" id\="org.eclipse.jdt.ui.text.codetemplates.fieldcomment" name\="fieldcomment"/><template autoinsert\="true" context\="methodcomment_context" deleted\="false" description\="Comment for non-overriding methods" enabled\="true" id\="org.eclipse.jdt.ui.text.codetemplates.met     hodcomment" name\="methodcomment">/**\n * ${tags}\n */</template><template autoinsert\="true" context\="overridecomment_context" deleted\="false" description\="Comment for overriding methods" enabled\="true" id\="org.eclipse.jdt.ui.text.codetemplates.overridecomment" name\="overridecomment">/* (non-Javadoc)\n * ${see_to_overridden}\n */</template><template      autoinsert\="true" context\="delegatecomment_context" deleted\="false" description\="Comment for delegate methods" enabled\="true" id\="org.eclipse.jdt.ui.text.codetemplates.delegatecomment" name\="delegatecomment">/**\n * ${tags}\n * ${see_to_target}\n */</template><template autoinsert\="false" context\="newtype_context" deleted\="false" description\="Newl     y created files" enabled\="true" id\="org.eclipse.jdt.ui.text.codetemplates.newtype" name\="newtype">/*\n * Copyright 2015 Palantir Technologies, Inc. All rights reserved.\n */\n\n${package_declaration}\n\n/**\n * \n */\n${typecomment}\n${type_declaration}</template><template autoinsert\="true" context\="classbody_context" deleted\="false" description\="Cod     e in new class type bodies" enabled\="true" id\="org.eclipse.jdt.ui.text.codetemplates.classbody" name\="classbody">\n</template><template autoinsert\="true" context\="interfacebody_context" deleted\="false" description\="Code in new interface type bodies" enabled\="true" id\="org.eclipse.jdt.ui.text.codetemplates.interfacebody" name\="interfacebody">\n</te     mplate><template autoinsert\="true" context\="enumbody_context" deleted\="false" description\="Code in new enum type bodies" enabled\="true" id\="org.eclipse.jdt.ui.text.codetemplates.enumbody" name\="enumbody">\n</template><template autoinsert\="true" context\="annotationbody_context" deleted\="false" description\="Code in new annotation type bodies" enabl     ed\="true" id\="org.eclipse.jdt.ui.text.codetemplates.annotationbody" name\="annotationbody">\n</template><template autoinsert\="false" context\="catchblock_context" deleted\="false" description\="Code in new catch blocks" enabled\="true" id\="org.eclipse.jdt.ui.text.codetemplates.catchblock" name\="catchblock">throw new RuntimeException(${exception_var});<     /template><template autoinsert\="false" context\="methodbody_context" deleted\="false" description\="Code in created method stubs" enabled\="true" id\="org.eclipse.jdt.ui.text.codetemplates.methodbody" name\="methodbody">${body_statement}</template><template autoinsert\="false" context\="constructorbody_context" deleted\="false" description\="Code in create     d constructor stubs" enabled\="true" id\="org.eclipse.jdt.ui.text.codetemplates.constructorbody" name\="constructorbody">${body_statement}</template><template autoinsert\="true" context\="getterbody_context" deleted\="false" description\="Code in created getters" enabled\="true" id\="org.eclipse.jdt.ui.text.codetemplates.getterbody" name\="getterbody">retur     n ${field};</template><template autoinsert\="true" context\="setterbody_context" deleted\="false" description\="Code in created setters" enabled\="true" id\="org.eclipse.jdt.ui.text.codetemplates.setterbody" name\="setterbody">${field} \= ${param};</template></templates>
```

**Comment by markelliot** [comment]
So some unfortunate things here:
1. the injection is not performed for Eclipse, so IntelliJ and Eclipse differ in this respect
2. the injected default header is not matched to the Checkstyle configuration (surely Gradle could do the format munging to produce a second template file)

**Comment by uschi2000** [comment]
Correct. Ideally, I'd like to remove all the copyright business from baseline and instead use https://github.com/hierynomus/license-gradle-plugin . This way, we'd loose IDE support, but it might be better overall?

**Comment by markelliot** [comment]
I think you're stuck with Baseline doing some kind of IDE configuration to make default-new-files have the right basic setup. Since Baseline is configuring that, it'd be convenient for it also to setup the Checkstyle configuration (or even if you move to recommending the license plugin, or applying it through Baseline, the license plugin) to match.

**Comment by ash211** [comment]
@uschi2000 was this fixed through the most recent series of three merged PRs, starting with #113 ?

**Comment by uschi2000** [comment]
Fixed.
